Documentation/coding-guidelines/project-guidelines.md

```diff
@@ -0,0 +1,116 @@
+These are optional for specific projects and don't require a full build pass but instead should be scoped to individual projects within the most appropriate full build pass.
+
+#Project configuration conventions
+For each unique configuration needed for a given library project a configuration property group should be added to the project so it can be selected and built in VS and also clearly identify the various configurations.<BR/>
+`<PropertyGroup Condition="'$(Configuration)|$(Platform)' == '$(<OSGroup>_<TargetGroup>_<ConfigurationGroup>)|$(Platform)'">`
+
+- `$(Platform) -> AnyCPU* | x86 | x64 | ARM | ARM64`
+- `$(Configuration) -> $(OSGroup)_$(TargetGroup)_$(ConfigurationGroup)`
+- `$(OSGroup) -> [Empty]* | Windows | Linux | OSX | FreeBSD`
+- `$(TargetGroup) -> [Empty]* | <TargetFrameworkMoniker> | <RuntimeIdentifier> | <Version> | <TFM><RID>`
+ - `$(PackageTargetFramework) -> netstandard | netcore50 | net46 | dnxcore50`
+ - `$(PackageTargetRuntime) -> aot`
+ - For more information on various targets see also [.NET Platform Standard](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/standard-platform.md)
+- `$(ConfigurationGroup) -> Debug* | Release`
+<BR/>`*` -> *default values*
+
+####*Examples*
+Project configurations for a pure IL library project which targets the defaults.
+```
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|AnyCPU'" />
+```
+Project configurations with a unique implementation for each OS
+```
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'FreeBSD_Debug|AnyCPU " />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'FreeBSD_Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Linux_Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Linux_Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'OSX_Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'OSX_Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Windows_Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Windows_Release|AnyCPU'" />
+```
+Project configurations that are unique for a few different target frameworks and runtimes
+```
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcore50aot_Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcore50aot_Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcore50_Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcore50_Release|AnyCPU'" />
```

Comment:
Sure, feel free to do it separately. I do think we will need the above for any partial facades that have more than one OS configuration though, right? For example, System.Runtime.Extensions has different builds for Windows, Linux, OSX, and FreeBSD, and we will also need to add the .NET Native configurations on top of that since its a partial facade.